File: AIiOS/AIiOS/Models/UserProfile.swift
import Foundation

struct UserProfile: Codable, Identifiable {
    var id: String          // Firebase UID — stored as both document ID and a field
    var username: String
    var displayName: String
    var bio: String
    var avatarURL: String?
    var phoneNumber: String?
    var createdAt: Date
    var updatedAt: Date
}

File: AIAndroid/app/src/main/kotlin/ai/ecoinference/app/models/UserProfile.kt
package ai.ecoinference.app.models

import com.google.firebase.Timestamp

data class UserProfile(
    val id: String = "",
    val username: String = "",
    val displayName: String = "",
    val bio: String = "",
    val avatarURL: String? = null,
    val phoneNumber: String? = null,
    val createdAt: Timestamp = Timestamp.now(),
    val updatedAt: Timestamp = Timestamp.now(),
)
